page/basics/loops/index.markdown

```diff
@@ -2,23 +2,30 @@
 
 #### foreach
 
-Traditional `for` loops exist but you almost never need them in Perl.
-`foreach` is usually simpler.
+C-like `for` loops exist but you almost never need them in Perl.
+`foreach` is usually simpler (but can be spelled as either `for`
+or `foreach`).
 
 Iterating over an array
 
-    my @array = (1, 2, 3, 4);
+    my @array = ('a', 'b', 'c', 'd');
 
-    foreach my $i (@array) {
-        print $i, "\n";
+    foreach my $e (@array) {
+        print $e, "\n";         # $e becomes each element of the array
+    }
+    # outputs "a\nb\nc\nd\n"
+
+    # exactly the same as above, only less typing :-)
+    for my $e (@array) {
+        print $e, "\n";
     }
 
 Iterating over a hash
 
     my %hash = (a => 1, b => 2, c => 3);
 
-    foreach my $i (keys %hash) {
-        print $hash{$i}, "\n";
+    for my $k (keys %hash) {
```

Comment:
Is there a reason you use `for` instead of `foreach` for most of the examples?  Or just personal preference?

Reply:
I guess it just is my personal preference.  I mean, I know it's my personal preference, I just didn't realize I was biasing the examples towards `for`.

I think people are going to see both "out in the wild"  though, so examples with both would be nice.